### PR TITLE
Wire-name labeling items are now useless for the nuke and syndie bombs

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Wire names on nuclear fission explosives and syndicate bombs will now be scrambled for anyone who tries to use a tool (or implant) that lets them see what each wire is.

## Why It's Good For The Game
It should really not be as easy to cheese syndicate bombs as it is, and it should ***really*** not be as easy to cheese defusing the nuclear bomb, especially on nukie rounds.

## Images of changes

https://user-images.githubusercontent.com/89928798/224180993-6c1dc342-1cda-49fe-aabc-0fb81a30b97e.mp4

## Testing
Loaded in, tried to defuse some bombs (and somehow went 2/2).

## Changelog
:cl:
tweak: Tools and implants that let you see wires will fail when trying to see the wires of syndicate bombs or the nuclear explosive.
/:cl:

### CODE REVIEW

File: code/datums/wires/wires.dm [hunk 1]
```diff
@@ -283,6 +283,47 @@
...
+			fake_wire_name += pick(fake_wire_chars)
+
+		return fake_wire_name
+
+	// even if you can see everything, duds should still be null
+	if(is_dud(wire))
+		return null
+	return wire
+
+	// can_see_wire_info(user) && !is_dud_color(color) ? get_wire(color) : null
```
Comment (S34NW) [inline code/datums/wires/wires.dm:324]:
```suggestion
	// can_see_wire_info(user) && !is_dud_color(color) ? get_wire(color) : null
```

??

File: code/datums/wires/wires.dm [hunk 2]
```diff
@@ -283,6 +283,47 @@
...
+ * * wire - a wire define.
+ */
+/datum/wires/proc/get_wire_name(mob/user, wire)
+	if(user.can_admin_interact())
+		// admins always get a free pass
+		return wire
+
+	var/can_probably_see_wires = FALSE
+	var/obj/item/held_item = user.get_active_hand()
+	if(istype(held_item) && HAS_TRAIT(held_item, TRAIT_SHOW_WIRE_INFO))
```
Comment (S34NW) [inline code/datums/wires/wires.dm:301]: How does this cope with things held in the off-hand after initially opening the wire interface with them? (open, swap hands, wire interact) does it remove the viewing of wires? or is it always there once the interface is opened?

Comment (lewcc) [reply code/datums/wires/wires.dm:301]: Good catch! It doesn't.
image
I'll make it check offhand as well.

File: code/datums/wires/wires.dm (hunk 1 above)
Comment (lewcc) [reply code/datums/wires/wires.dm:324]: oops, I pulled that down so I didn't forget the checks. I'll remove that

**Review by S34NW — APPROVED** [review]

File: code/__HELPERS/traits.dm [hunk 3]
```diff
@@ -206,6 +206,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 //***** ITEM TRAITS *****//
 #define TRAIT_BUTCHERS_HUMANS "butchers_humans"
 #define TRAIT_CMAGGED "cmagged"
+/// Wires on this will have their titles randomized for those with SHOW_WIRES
+#define TRAIT_OBSCURED_WIRES "obscured_wires"
```
Comment (GDNgit) [inline code/__HELPERS/traits.dm:210]: probably wanna add this to the global trait list so admemes can select it

Comment (lewcc) [reply code/__HELPERS/traits.dm:210]: Done

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**